src/game_battlealgorithm.cpp

```diff
@@ -815,6 +816,10 @@ bool Game_BattleAlgorithm::AlgorithmBase::OriginalTargetsSet() const {
 	return (original_targets.size() > 0);
 }
 
+Game_Battler* Game_BattleAlgorithm::AlgorithmBase::GetFirstOriginalTarget() const {
+	return *original_targets.begin();
```

Comment:
If you're going to return a pointer, check for empty first and return null if no original targets. Otherwise, return a `Game_Battler&` and add an `assert(!original_targets.empty())`

Reply:
This is probably a dumb question but should I still make the change what this function returns? Because you have successfully made tests and I'm afraid I will break something if I try to change this. And maybe I missed something but I don't quite understand the assert suggestion. For me it looks like an assertion gets triggered when original targets are set.

Reply:
This is a very minor change, and converting pointer to reference any bugs will be compiler errors.

Just be sure to get the assert correct. `assert(!original_targets.empty())` means crash the program if I call this function and the targets are empty. asserts are only enabled in debug mode so you can quickly test a debug build.

Reply:
Currently your code will crash when there are no ``original_targets``. This means you make an assumption here that this is not supposed to happen. An ``assert`` is nice to mark impossible situations - When the assert is triggered it means the code has a bug. (note that user input should never trigger asserts, for user generated problems there should be more sane error handling)

Using ``&`` instead of ``*`` here makes sense because this indicates to the caller "The return value is always a valid ``Battler``". For ``*`` the caller must expect ``NULL/nullptr``.

Reply:
I somehow could not think straight for the last few hours. But now I went for the first variant in @fmatthew5876 's comment, returning a pointer and doing the check for empty. I have pushed a new commit.